feat: add increase governance delay time function

evandrosaturnino opened the pull request:
## Description 

- [X] - Add increase governance time delay in thusd contract
- [X] - Update the contract tests reflecting the governance time delay dynamics changes
- [X] - Update the deployment scripts in lib ethers supporting BOB's testnet and mainnet

## Notice 

- [x] Have you checked to ensure there aren't other open Pull Request for the same update/change? 

## Pull Request Type 

- [x] Feat update (Breaking/Non-breaking Change) 

## Testing Please outline all testing steps 

Testing locally: 
1. Clone the repo locally: `git clone https://github.com/Threshold-USD/dev.git` 
2. Install the dependecies with yarn: `yarn` 
3. Run the app: `yarn start-dev-frontend`

evandrosaturnino committed:
feat: add bob testnet deployments

evandrosaturnino committed:
feat: update governance delay time in contracts

evandrosaturnino committed:
refactor: update new governance delay time param name

evandrosaturnino committed:
feat: update tests with the governance delay time new dynamics

evandrosaturnino committed:
feat: update deployment scripts to support bob layers

evandrosaturnino committed:
chore: update abis and types automatically generated

vzotova commented on packages/contracts/contracts/THUSDToken.sol:
```diff
@@ -94,8 +93,6 @@ contract THUSDToken is Ownable, CheckContract, ITHUSDToken {
         _CACHED_CHAIN_ID = block.chainid;
         _CACHED_DOMAIN_SEPARATOR = _buildDomainSeparator(_TYPE_HASH, hashedName, hashedVersion);
-        governanceTimeDelay = _governanceTimeDelay;
-        require(governanceTimeDelay <= 30 weeks, "Governance delay is too big");
```
I'd keep it in the constructor anyway, otherwise will be period of time when delay is zero

evandrosaturnino replied:
Addressed in [refactor: add governance time delay in thusd constructor](https://github.com/Threshold-USD/dev/pull/113/commits/9f771a49a984e3122ea25dd5881305289819e1f9)

vzotova commented on packages/contracts/contracts/PCV.sol:
```diff
@@ -179,6 +175,8 @@ contract PCV is IPCV, Ownable, CheckContract, SendCollateral {
         require(_council != council || _treasury != treasury, "PCV: these roles already set");
 
         changingRolesInitiated = block.timestamp;
+        governanceTimeDelay = thusdToken.getGovernanceTimeDelay();
```
I'd suggest to have this as independent delay, same as in token contract

evandrosaturnino replied:
Addressed in [refactor: add governance time delay in pcv constructor](https://github.com/Threshold-USD/dev/pull/113/commits/273a76adde2a0a3de6774338fedec2b873578bd4)

vzotova commented on packages/contracts/contracts/THUSDToken.sol:
```diff
@@ -240,6 +253,12 @@ contract THUSDToken is Ownable, CheckContract, ITHUSDToken {
+    function getGovernanceTimeDelay() external view override returns (uint256) {
+        return governanceTimeDelay;
+    }
+
```
`governanceTimeDelay` is already `public` which means there is getter for that field

evandrosaturnino replied:
Addressed in [refactor: add governance time delay in thusd constructor](https://github.com/Threshold-USD/dev/pull/113/commits/9f771a49a984e3122ea25dd5881305289819e1f9)

evandrosaturnino committed:
refactor: add governance time delay in pcv constructor

evandrosaturnino committed:
refactor: add governance time delay in thusd constructor

evandrosaturnino committed:
refactor: update tester contracts with the governance time delay

evandrosaturnino committed:
refactor: update tests reflecting governance delay modifications

evandrosaturnino committed:
refactor: update deploy scripts reflecting governance delay modifications

evandrosaturnino committed:
chore: automatic abi and types generation

evandrosaturnino committed:
style: lint fixes

evandrosaturnino committed:
chore: update contracts on bob testnet

vzotova reviewed (approved):
LGTM
just one question: do we want same `increase` method for PCV as well?

evandrosaturnino committed:
layout: add extra info about base rate fees

evandrosaturnino commented:
> LGTM just one question: do we want same `increase` method for PCV as well?

Thank you for the review, @vzotova!

I can't think a reason why it should. The reason why we added this feature in thUSD is because we want to deploy it with a decreased amount initially to provide some agility on adding new collaterals BTC and ETH backed assets, then we can increase it just before L1 fungibility to mitigate any risk with the existing thUSD on mainnet.

Can't see why the increase delay governance time functionality should be also added in PCV, since it can already start with 120 days. (30 days more than the existing 90 days on mainnet)
